### PR TITLE
Add metrics handler

### PR DESCRIPTION
**Pull request by SuperQ**
Add simple Prometheus client metrics handler.

### CODE REVIEW

**Comment by SuperQ** [comment]
I haven't tested this at all, but it should make the app respond to some basic stuff on `/metrics`.

File: donate/app.py [hunk 1]
```diff
@@ -29,6 +30,11 @@ def create_app(config_object=ProdConfig):
 
     app.get_stripe_key = get_stripe_key
 
+    # Handle metrics requests.
+    @app.route("/metrics")
+    def metrics():  
+        return Response(prometheus_client.generate_latest(), mimetype=prometheus_client.CONTENT_TYPE_LATEST)
```
Comment (marcidy) [inline donate/app.py]: Where is Response supposed to come from?  failing to find it

Comment (SuperQ) [reply donate/app.py]: Flask thing I forgot to add.

**Comment by marcidy** [comment]
LGTM:

sample output:
# HELP python_gc_objects_collected_total Objects collected during gc
# TYPE python_gc_objects_collected_total counter
python_gc_objects_collected_total{generation="0"} 30233.0
python_gc_objects_collected_total{generation="1"} 5277.0
python_gc_objects_collected_total{generation="2"} 138.0
# HELP python_gc_objects_uncollectable_total Uncollectable object found during GC
# TYPE python_gc_objects_uncollectable_total counter
python_gc_objects_uncollectable_total{generation="0"} 0.0
python_gc_objects_uncollectable_total{generation="1"} 0.0
python_gc_objects_uncollectable_total{generation="2"} 0.0
# HELP python_gc_collections_total Number of times this generation was collected
# TYPE python_gc_collections_total counter
python_gc_collections_total{generation="0"} 262.0
python_gc_collections_total{generation="1"} 23.0
python_gc_collections_total{generation="2"} 2.0
# HELP python_info Python platform information
# TYPE python_info gauge
python_info{implementation="CPython",major="3",minor="5",patchlevel="3",version="3.5.3"} 1.0
# HELP process_virtual_memory_bytes Virtual memory size in bytes.
# TYPE process_virtual_memory_bytes gauge
process_virtual_memory_bytes 3.73456896e+08
# HELP process_resident_memory_bytes Resident memory size in bytes.
# TYPE process_resident_memory_bytes gauge
process_resident_memory_bytes 7.1241728e+07
# HELP process_start_time_seconds Start time of the process since unix epoch in seconds.
# TYPE process_start_time_seconds gauge
process_start_time_seconds 1.56210461953e+09
# HELP process_cpu_seconds_total Total user and system CPU time spent in seconds.
# TYPE process_cpu_seconds_total counter
process_cpu_seconds_total 1.1400000000000001
# HELP process_open_fds Number of open file descriptors.
# TYPE process_open_fds gauge
process_open_fds 8.0
# HELP process_max_fds Maximum number of open file descriptors.
# TYPE process_max_fds gauge
process_max_fds 1024.0

**Comment by SuperQ** [comment]
We can instrument additional things, see the [official docs](https://github.com/prometheus/client_python#instrumenting).

We might want to add a route restriction in Caddy so only monitoring can get to it. But it's mostly harmless to see this stuff publicly.

**Pull request merged**